src/ray/raylet/node_manager.cc

```diff
@@ -1494,11 +1487,29 @@ void NodeManager::DispatchScheduledTasksToWorkers() {
   while (!tasks_to_dispatch_.empty()) {
     auto task = tasks_to_dispatch_.front();
     auto reply = task.first;
+    auto spec = task.second.GetTaskSpecification();
     std::shared_ptr<Worker> worker =
-        worker_pool_.PopWorker(task.second.GetTaskSpecification());
+        worker_pool_.PopWorker(spec);
     if (worker == nullptr) {
       return;
     }
+
+    bool schedulable = new_resource_scheduler_->SubtractNodeAvailableResources(
+       client_id_.Binary(), spec.GetRequiredResources().GetResourceMap());
+    if (schedulable == false) {
+      return;
+    }
+    // Handle the allocation to specific resource IDs.
+    auto acquired_resources =
+          local_available_resources_.Acquire(spec.GetRequiredResources());
```

Comment:
It seems unfortunate that we have resources tracked in two places. Can we instead have the new scheduler track the resource ids internally?

Reply:
You are correct. Will do this in the next PR.